components/org.wso2.carbon.identity.login.resolver/org.wso2.carbon.identity.login.resolver.regex/src/main/java/org/wso2/carbon/identity/login/resolver/regex/RegexLoginResolver.java

```diff
@@ -0,0 +1,204 @@
+                if (StringUtils.isBlank(regex)) {
+                    continue;
+                }
+                Pattern pattern = Pattern.compile(regex);
+                if (pattern.matcher(loginIdentifier).matches()) {
+                    authenticationResult = userStoreManager.
+                            authenticateWithID(claimURI, loginIdentifier, credential, StringUtils.EMPTY);
+                    if (AuthenticationResult.AuthenticationStatus.SUCCESS.
+                            equals(authenticationResult.getAuthenticationStatus())) {
+                        break;
+                    }
+                }
+            }
+            /*
+            If allowed attributes has only username claim, get authenticationResult even if
+            the username claim has no configured regex pattern.
+             */
+            if (allowedAttributes.size() == 1 && allowedAttributes.contains(
+                    UserCoreClaimConstants.USERNAME_CLAIM_URI)) {
+                authenticationResult =
+                        userStoreManager.authenticateWithID(UserCoreClaimConstants.USERNAME_CLAIM_URI, loginIdentifier,
+                                credential, StringUtils.EMPTY);
+            }
+        } catch (UserStoreException e) {
+            log.error("An error occurred while resolving authentication result.", e);
+        }
+        return authenticationResult;
+    }
+
+    /**
+     * Sets the resolved user result object with the user details if a valid user is found.
+     *
+     * @param userStoreManager   The user store manager.
+     * @param claimURI           The claim URI which should be used to extract the user.
+     * @param loginIdentifier    The login identifier provided by the user.
+     * @param resolvedUserResult The dummy ResolvedUserResult object for which the user details should be added to.
+     * @param claim              The claim object.
+     * @throws org.wso2.carbon.user.core.UserStoreException If there is an error while extracting a list of users with
+     *                                                      the ID.
+     */
+    private void setResolvedUserResult(UniqueIDUserStoreManager userStoreManager, String claimURI,
+                                       String loginIdentifier, ResolvedUserResult resolvedUserResult, Claim claim)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        List<User> userList = userStoreManager.getUserListWithID(claimURI, loginIdentifier, null);
+        if (userList.size() == 1) {
+            resolvedUserResult.setResolvedStatus(ResolvedUserResult.UserResolvedStatus.SUCCESS);
+            resolvedUserResult.setResolvedClaim(claimURI);
+            resolvedUserResult.setResolvedValue(loginIdentifier);
+            User user = userList.get(0);
+            user.setUsername(user.getDomainQualifiedUsername());
+            resolvedUserResult.setUser(user);
+        } else if (userList.size() > 1) {
```

Comment:
What If UserList.size() is 0?

Reply:
A matching user could not be found with the provided information. Again, we are sending an `UserResolvedStatus.FAIL` status in the `resolvedUserResult` object. As, an improvement we can set an error message + error code to the `resolvedUserResult` object.

Reply:
+1

Reply:
Fixed with https://github.com/wso2-extensions/identity-governance/pull/652/commits/8850e6ed66ee47b980c57d32cb7918c9dde6d577.